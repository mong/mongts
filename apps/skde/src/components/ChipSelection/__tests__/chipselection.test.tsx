import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import { ChipSelection } from "../.";
import { vi, describe, test, expect } from "vitest";

vi.mock("next/font/google", () => ({
  Plus_Jakarta_Sans: () => ({
    style: {
      fontFamily: "mocked",
    },
  }),
}));

describe("ChipSelection", () => {
  test("renders with correct labels and help text", () => {
    const { getByText, getByRole, container } = render(
      <ChipSelection
        leftChipLabel="Chip 1"
        rightChipLabel="Chip 2"
        leftChipHelpText="Help text for Chip 1"
        rightChipHelpText="Help text for Chip 2"
        state={false}
        stateSetter={() => {}}
        trueChip="left"
        hoverBoxOffset={[0, 0]}
        hoverBoxPlacement="top"
        hoverBoxMaxWidth={100}
      />,
    );

    expect(getByText("Chip 1")).toBeTruthy();
    expect(getByText("Chip 2")).toBeTruthy();
    const button1 = getByRole("button", { name: "Help text for Chip 1" });
    expect(button1).toBeTruthy();
    const button2 = getByRole("button", { name: "Help text for Chip 2" });
    expect(button2).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test("toggles state when a chip is clicked", () => {
    const setState = vi.fn();
    render(
      <ChipSelection
        leftChipLabel="Chip 1"
        rightChipLabel="Chip 2"
        leftChipHelpText="Help text for Chip 1"
        rightChipHelpText="Help text for Chip 2"
        state={false}
        stateSetter={setState}
        trueChip="left"
        hoverBoxOffset={[0, 0]}
        hoverBoxPlacement="top"
        hoverBoxMaxWidth={100}
      />,
    );

    fireEvent.click(screen.getByText("Chip 1"));
    expect(setState).toHaveBeenCalledWith(true);

    fireEvent.click(screen.getByText("Chip 2"));
    expect(setState).toHaveBeenCalledWith(false);
  });
  test("renders with trueChip right", () => {
    const { container } = render(
      <ChipSelection
        leftChipLabel="Chip 1"
        rightChipLabel="Chip 2"
        leftChipHelpText="Help text for Chip 1"
        rightChipHelpText="Help text for Chip 2"
        state={false}
        stateSetter={() => {}}
        trueChip="right"
        hoverBoxOffset={[0, 0]}
        hoverBoxPlacement="top"
        hoverBoxMaxWidth={100}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
