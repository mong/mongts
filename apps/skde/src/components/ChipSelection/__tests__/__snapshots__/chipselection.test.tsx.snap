// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChipSelection > renders with correct labels and help text 1`] = `
<div>
  <div
    class="MuiStack-root css-14g5whm-MuiStack-root"
  >
    <div
      aria-label="Help text for Chip 1"
      class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-outlinedPrimary css-s419k8-MuiButtonBase-root-MuiChip-root"
      data-mui-internal-clone-element="true"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
      >
        Chip 1
      </span>
    </div>
    <div
      aria-label="Help text for Chip 2"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-1pymu8d-MuiButtonBase-root-MuiChip-root"
      data-mui-internal-clone-element="true"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Chip 2
      </span>
    </div>
  </div>
</div>
`;

exports[`ChipSelection > renders with trueChip right 1`] = `
<div>
  <div
    class="MuiStack-root css-14g5whm-MuiStack-root"
  >
    <div
      aria-label="Help text for Chip 1"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-1pymu8d-MuiButtonBase-root-MuiChip-root"
      data-mui-internal-clone-element="true"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Chip 1
      </span>
    </div>
    <div
      aria-label="Help text for Chip 2"
      class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-outlinedPrimary css-s419k8-MuiButtonBase-root-MuiChip-root"
      data-mui-internal-clone-element="true"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
      >
        Chip 2
      </span>
    </div>
  </div>
</div>
`;
